modules/ui/src/main/scala/lucuma/ui/table/TableHooks.scala

```diff
@@ -28,32 +28,25 @@ private object TableHooks:
   private object PrefsLoaded extends NewType[Boolean]
   private type PrefsLoaded = PrefsLoaded.Type
 
-  private object CanSave extends NewType[Boolean]
-  private type CanSave = CanSave.Type
-
   private def hook[T] =
     CustomHook[TableOptionsWithStateStore[DefaultA, T]]
       .useReactTableBy(_.tableOptions)
       .useState(PrefsLoaded(false))
-      .useRef(CanSave(false))
-      .useEffectOnMountBy((props, table, prefsLoadad, canSave) =>
+      .useEffectOnMountBy((props, table, prefsLoadad) =>
         (props.stateStore.load() >>=
           (mod => table.modState(mod).to[DefaultA]))
           .guarantee( // This also forces a rerender, which react-table isn't doing by just changing the state.
             prefsLoadad.setStateAsync(PrefsLoaded(true))
           )
       )
-      .useEffectWithDepsBy((_, table, _, _) => table.getState())((props, _, prefsLoadad, canSave) =>
```

Comment:
According to this comment, the `CanSave` logic is to avoid saving preferences as soon as we load them, since it waits an extra render cycle.

But I see how this would break if `load()` fails. We can probably fix the bug by keeping the mechanism and adding a `canSave.setAsync(CanSave(true))` in an error handler for the `load()` above.

Reply:
If that's the case shouldn't we set the can save right after load?
The result of the current logic is that the first time we change e.g. sorting it doesn't save it

Reply:
Remember that effects are run after a render.

The current logic is: `Prefs loaded and set into state` => `render` => `useEffectWithDepsBy run, prefs not saved, but CanSave set to true so they can be saved from now on`. If we set right after load (or we remove the `CanSave` logic) then they will be saved in that first run of `useEffectWithDepsBy`, which we probably don't want.

The first change should be saved if `load()` completed successfully, but it would be skipped if it failed, that's why we should set `CanSave` to `true` in an error handler. Or is the first save failing even when `load()` completes successfully?

Reply:
Yes the first save fails even if load is ok

Reply:
I looked into this and it seems that react-table changed its behavior, before `modState` didn't trigger and rerender and now it does. Therefore, the first rerender was sometimes being triggered when `prefsLoadad` was still false, breaking the whole mechanism.

I'm submitting another PR contemplating this while still avoiding saving prefs as soon as they are loaded.

Reply:
https://github.com/gemini-hlsw/lucuma-ui/pull/844